app/javascript/src/Settings/components/Common/FormCollection.jsx

```diff
@@ -0,0 +1,24 @@
+// @flow
+
+import * as React from 'react'
+import { FormFieldset, FormLegend } from 'Form'
+import type { FormSettingsProps } from 'Settings/types'
+
+type Props = {
+  collection: FormSettingsProps[],
+  ItemComponent: (FormSettingsProps) => React.Element<empty>,
+  legend: string
+}
+
+const FormCollection = ({collection, ItemComponent, legend}: Props) => {
+  return (
+    <FormFieldset id={`fieldset-${legend.replace(/\s+/g, '')}`}>
+      <FormLegend>{legend}</FormLegend>
+      { collection.map((itemProps, i) => <ItemComponent {...itemProps} key={i} />) }
```

Comment:
We should avoid using index as keys:
```suggestion
      { collection.map(itemProps => <ItemComponent {...itemProps} key={itemProps.name} />) }
```